uwblueprint/sistering
# Add User management table

## Ticket link

Closes #454, #455, #540



## Implementation description
* Added user management table as designed via tanstack react-table
* Integrated search and pagination to table
* Integrated existing profile drawer to table (still need to hook up profile drawer to interact with API)
* Added new backend resolvers to update/append branches only, due to limited input in user management table
* All branch edit actions are fully implemented and integrated
* Full user management flow should be complete
* 
https://user-images.githubusercontent.com/44826218/182489354-86298682-6e56-4051-8409-3585fd8c5099.mp4

### Test queries for new resolvers

```gql
mutation {
  appendBranchesForMultipleUsersByEmail(emails: ["sistering+employee@uwblueprint.org", "sistering+employee1@uwblueprint.org"], branchIds: [1])
}
```


```gql
mutation {
  updateUserBranchesByEmail(email: "sistering+employee@uwblueprint.org", branchIds: [1])
}
```


## Steps to test
1. Login as a admin and go to `/admin/users` and interact with table. It should work as per designed


## What should reviewers focus on?
* Code cleanliness (will do further follow up in full integration ticket #540 )
* Code correctness

## Checklist
